### ISSUE TITLE
GitHub webhook to allow automatic uploading of docs after pushing a tag

### ISSUE THREAD

**Issue by hdgarrood** [purescript/pursuit#118]
We should do #96 first.

**Comment by paf31** [comment]
Will Pursuit reject the upload if the tag already exists?

**Comment by hdgarrood** [comment]
No, it just overwrites the current one. I was considering making it issue a warning if the version you're uploading already exists.

**Comment by hdgarrood** [comment]
Although it's not obvious how to warn non-human API clients...

**Comment by paf31** [comment]
We probably should make it an error?

**Comment by hdgarrood** [comment]
Would it then be impossible to replace existing versions of things?

**Comment by paf31** [comment]
Yes, but for now we can just delete things by hand if necessary. I don't know which option is better.

**Comment by hdgarrood** [comment]
People (including me) have deleted tags or retagged different commits on PureScript packages, though - I think we ought to be at least as flexible as bower/GitHub is.

Perhaps the web UI could warn, and the API endpoint could require an `X-Replace-Existing` header or something?

**Comment by paf31** [comment]
How about the API returns an error response, but you can force an override with `?force`?

**Comment by hdgarrood** [comment]
Yeah, that works.

**Comment by paf31** [comment]
I just would like to make it a conscious decision to overwrite something. Maybe the web app could even have an extra step if an overwrite is needed?

**Comment by hdgarrood** [comment]
Sounds good.

**Comment by hdgarrood** [comment]
We have http://taylor.fausak.me/2015/11/15/automatically-generate-purescript-documentation/ now so I think automatic uploading is solved. I'll change the title of this issue re overwriting.
